analytic_engine/src/lib.rs

```diff
@@ -288,30 +288,41 @@ pub struct KafkaWalConfig {
     pub meta_namespace: MessageQueueWalConfig,
 }
 
-/// Config for wal based on RocksDB
-#[derive(Debug, Clone, Deserialize, Serialize)]
-#[serde(default)]
-pub struct RocksDBConfig {
-    /// Data directory used by RocksDB.
-    pub data_dir: String,
+#[cfg(feature = "wal-rocksdb")]
+pub use rocksdb::*;
 
-    pub data_namespace: RocksDBWalConfig,
-    pub meta_namespace: RocksDBWalConfig,
-}
+#[cfg(feature = "wal-rocksdb")]
+mod rocksdb {
+    use serde::{Deserialize, Serialize};
+    use wal_rocksdb::config::Config as RocksDBWalConfig;
 
-impl Default for RocksDBConfig {
-    fn default() -> Self {
-        Self {
-            data_dir: "/tmp/ceresdb".to_string(),
-            data_namespace: Default::default(),
-            meta_namespace: Default::default(),
+    /// Config for wal based on RocksDB
+    #[derive(Debug, Clone, Deserialize, Serialize)]
+    #[serde(default)]
+    pub struct RocksDBConfig {
+        /// Data directory used by RocksDB.
+        pub data_dir: String,
+
+        pub data_namespace: RocksDBWalConfig,
+        pub meta_namespace: RocksDBWalConfig,
+    }
+
+    impl Default for RocksDBConfig {
+        fn default() -> Self {
+            Self {
+                data_dir: "/tmp/ceresdb".to_string(),
+                data_namespace: Default::default(),
+                meta_namespace: Default::default(),
+            }
         }
     }
 }
+
 /// Options for wal storage backend
 #[derive(Debug, Clone, Deserialize, Serialize)]
 #[serde(tag = "type")]
 pub enum WalStorageConfig {
```

Comment:
But this enum cannot be moved to any specific `wal-`impl, not suitable for `wal` also.